File: quixo/server/src/services/cart/cart.model.js
const mongoose = require('mongoose');

const cartItemSchema = new mongoose.Schema({
  productId: { type: mongoose.Schema.Types.ObjectId, ref: 'Product', required: true },
  name: { type: String, required: true },
  image: String,
  price: { type: Number, required: true },
  qty: { type: Number, default: 1, min: 1 },
  type: { type: String, enum: ['quick', 'ecommerce'], default: 'quick' }
}, { _id: true });

const cartSchema = new mongoose.Schema({
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true, unique: true },
  items: [cartItemSchema],
}, { timestamps: true });

cartSchema.virtual('total').get(function () {
  return this.items.reduce((sum, item) => sum + (item.price * item.qty), 0);
});

cartSchema.set('toJSON', { virtuals: true });

module.exports = mongoose.model('Cart', cartSchema);

File: quixo/flutter/mart_customerapp/martcustomerapp-main/lib/features/cart/data/cart_model.dart
import 'package:equatable/equatable.dart';

class CartAddRequestModel extends Equatable {
  final String productId;
  final int number;

  const CartAddRequestModel({
    required this.productId,
    required this.number,
  });

  Map<String, dynamic> toJson() {
    return {
      "product_id": productId,
      "number": number,
    };
  }

  @override
  List<Object?> get props => [productId, number];
}